### DISCUSSION TITLE
Performing area selection in Cypress

### DISCUSSION THREAD

**Discussion by madushajg** [cypress-io/cypress#16945] [Questions and Help]
I have a requirement to test a zoom-in functionality for a selected area like depicted below.

![area_selection](https://user-images.githubusercontent.com/26219651/122075233-896b1c80-ce17-11eb-87fc-5c2c3dfc7ad6.gif)

Here I need to perform the selection over the graph. I referred to many available discussions and suggestions, but couldn't get any success.

I tried to achieve this as a [drag n drop](https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/testing-dom__drag-drop/cypress/integration/drag_n_drop_spec.js) solution too. But didn't work.

```typescript
  cy.get('[data-testid="graph"]')
      .trigger('mousedown', { pageX: 100, pageY: 120 })
      .trigger('mousemove', { pageX: 300, pageY: 120 })
      .trigger('mouseup')
```

How can I achieve this using Cypress?

**Comment by PaoloKa** [comment]
I used this and seems to do the trick
```
  cy.get(".canvas-container")
    .trigger("mousedown", "bottomRight")
    .trigger("mousemove")
    .trigger("mouseup", "center");
```

**Reply by tar04** [reply]
Hey @PaoloKa ! How did you make it work? Cause i have pretty simmilar situation but instead of canvas i need to select inside div block

**Comment by goska-malaga** [comment]
this is answered in https://github.com/cypress-io/cypress/issues/3441
it worked for me to use  cy.window().then((win) => {
I just needed to do mousemove on different element then mousedown
